polyfill/test/validStrings.mjs

```diff
@@ -383,9 +383,15 @@ const timeSpecWithOptionalTimeZoneNotAmbiguous = choice(
 );
 const timeSpecSeparator = seq(dateTimeSeparator, timeSpec);
 
-const dateSpecMonthDay = seq(['--'], dateMonth, ['-'], dateDay);
+function validateDayOfMonth(result, { year, month, day }) {
+  if (day > ES.ISODaysInMonth(year, month)) throw SyntaxError('retry if bad day of month');
```

Comment:
Shouldn't this be `RangeError` instead of `SynaxError`?

Reply:
Ah nevermind -- I forgot that we throw `SyntaxError`s in the fuzzer for rejecting syntactic "out-of-bounds" errors.